drain3/template_miner.py

```diff
@@ -44,13 +44,21 @@ def __init__(self,
         self.config = config
 
         self.profiler: Profiler = NullProfiler()
+
         if self.config.profiling_enabled:
             self.profiler = SimpleProfiler()
 
         self.persistence_handler = persistence_handler
 
         param_str = self.config.mask_prefix + "*" + self.config.mask_suffix
-        self.drain = Drain(
+
+        # Follow the configuration in the configuration file to instantiate Drain
+        target_obj = self.config.matched_pattern
+        if target_obj not in ["Drain","JaccardDrain"]:
+            raise ValueError(f"Invalid matched_pattern: {target_obj}, must be either 'Drain' or 'JaccardDrain'")
+
+        '''
+        self.drain = JaccardDrain(
```

Comment:
Why did you make JaccardDrain the default? Default  behavior should stay exactly as it is. JaccardDrain should be opt-in.

Reply:
You are right, Drain should be the default, this is the point I missed in code.